### PR TITLE
refactor: move function hosted notification code

### PR DESCRIPTION
**Pull request by a1exwang**
As the offline discussion with @eriolchan team, we will move the function hosted notification code into a inherited class of `TeamsBotImpl`

https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/13814055

### CODE REVIEW

**Comment by a1exwang** [comment]
E2E TEST: n/a

File: packages/fx-core/src/plugins/resource/bot/functionsHostedBot/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (swatDong) [inline packages/fx-core/src/plugins/resource/bot/functionsHostedBot/plugin.ts:1]: Add code owners for `functionHostedBot/`, to include both teams' members.

Comment (a1exwang) [reply packages/fx-core/src/plugins/resource/bot/functionsHostedBot/plugin.ts:1]: Added

**Review by swatDong — APPROVED** [review]

**Review by Siglud — APPROVED** [review]
LGTM

**Pull request merged**